doc/pipeline-01.yaml

```diff
@@ -0,0 +1,63 @@
+name: mlflow-sklearn-e2e
+description: |
+  End-to-end pipeline template that takes in an MLFlow compatible codeset,
+  runs the MLFlow project to train a model, then creates a KFServing prediction
+  service that can be used to run predictions against the model."
+inputs:
+  - name: mlflow-codeset
+    description: an MLFlow compatible codeset
+    type: codeset
+  - name: predictor
+    description: type of predictor engine
+    type: string
+    default: auto
+outputs:
+  - name: prediction-url
+    description: "The URL where the exposed prediction service endpoint can be contacted to run predictions."
+    type: string
+steps:
+  - name: builder
+    image: ghcr.io/fuseml/mlflow-builder:1.0
+    inputs:
+      - codeset:
+          name: '{{ inputs.mlflow-codeset }}'
+          path: /project
+    outputs:
+      - name: mlflow-env
+        image:
+          dockerfile: '/project/.fuseml/Dockerfile'
+          name: 'registry.fuseml-registry/mlflow-builder/{{ inputs.mlflow-codeset.name }}:{{ inputs.mlflow-codeset.version }}'
```

Comment:
Here is codeset name referred as `inputs.mlflow-codeset.name`, but in the input section it's just `inputs.mlflow-codeset`. Why the difference?

Reply:
In the input section it is declaring a codeset type of input, while here it is a reference to one of the fields available from the codeset (`name`)

Reply:
Why is the image part actually mentioned in the `outputs`? It does not seem any other step needs it, __trainer__ step is able to read it just when referencing `mlflow-env`... I mean couldn't the image location actually be internal, not exposed in the workflow description?

Reply:
The image is in the outputs because it is the expected output from the step.

Note that the name of the image also includes the registry where it will be stored, removing it also removes the option to store the image on another registries.

Reply:
I'm not saying it should not be possible to set up the destination registry, I was just thinking, in order to have minimal working example, specifying image name and location could actually be optional